drivers/gpio/gpio_mcux_igpio.c

```diff
@@ -102,6 +96,98 @@ static int mcux_igpio_read(struct device *dev,
 	return 0;
 }
 
+static int mcux_igpio_port_get_raw(struct device *dev, u32_t *value)
+{
+	const struct mcux_igpio_config *config = dev->config->config_info;
+	GPIO_Type *base = config->base;
+
+	*value = base->DR;
+
+	return 0;
+}
+
+static int mcux_igpio_port_set_masked_raw(struct device *dev, u32_t mask,
+		u32_t value)
+{
+	const struct mcux_igpio_config *config = dev->config->config_info;
+	GPIO_Type *base = config->base;
+
+	base->DR = (base->DR & ~mask) | (mask & value);
+
+	return 0;
+}
+
+static int mcux_igpio_port_set_bits_raw(struct device *dev, u32_t mask)
+{
+	const struct mcux_igpio_config *config = dev->config->config_info;
+	GPIO_Type *base = config->base;
+
+	base->DR_SET = mask;
+
+	return 0;
+}
+
+static int mcux_igpio_port_clear_bits_raw(struct device *dev, u32_t mask)
+{
+	const struct mcux_igpio_config *config = dev->config->config_info;
+	GPIO_Type *base = config->base;
+
+	base->DR_CLEAR = mask;
+
+	return 0;
+}
+
+static int mcux_igpio_port_toggle_bits(struct device *dev, u32_t mask)
+{
+	const struct mcux_igpio_config *config = dev->config->config_info;
+	GPIO_Type *base = config->base;
+
+	base->DR_TOGGLE = mask;
+
+	return 0;
+}
+
+static int mcux_igpio_pin_interrupt_configure(struct device *dev,
+		unsigned int pin, enum gpio_int_mode mode,
+		enum gpio_int_trig trig)
+{
+	const struct mcux_igpio_config *config = dev->config->config_info;
+	struct mcux_igpio_data *data = dev->driver_data;
+	GPIO_Type *base = config->base;
+	u8_t icr;
+	int shift;
+
+	if ((mode == GPIO_INT_MODE_EDGE) && (trig == GPIO_INT_TRIG_LOW)) {
+		icr = 3;
+	} else if ((mode == GPIO_INT_MODE_EDGE) &&
+		   (trig == GPIO_INT_TRIG_HIGH)) {
+		icr = 2;
+	} else if ((mode == GPIO_INT_MODE_LEVEL) &&
+		   (trig == GPIO_INT_TRIG_HIGH)) {
+		icr = 1;
+	} else {
+		icr = 0;
+	}
+
+	if (pin < 16) {
+		shift = 2 * pin;
+		base->ICR1 = (base->ICR1 & ~(3 << shift)) | (icr << shift);
+	} else if (pin < 32) {
+		shift = 2 * (pin - 16);
+		base->ICR2 = (base->ICR2 & ~(3 << shift)) | (icr << shift);
+	} else {
```

Comment:
There is an assert in the API that guarantees that pin number is smaller than the maximum port width. Since the maximum port width in case of mcux driver is 32 we could skip the `else`.